core/src/indexer/start.rs

```diff
@@ -68,39 +70,214 @@ pub enum StartIndexingError {
+                event_name.to_string(),
+                end_block,
+                latest_block,
+            ));
+        }
+    }
+
+    let last_known_start_block = if manifest_start_block.is_some() {
+        let last_synced_block = get_last_synced_block_number(config).await;
+
+        if let Some(value) = last_synced_block {
+            let start_from = value + 1;
+            info!(
+                "{} Found last synced block number - {:?} rindexer will start up from {:?}",
+                event_name, value, start_from
+            );
+            Some(start_from)
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+
+    let start_block =
+        last_known_start_block.unwrap_or(manifest_start_block.unwrap_or(latest_block));
+    let end_block = std::cmp::min(manifest_end_block.unwrap_or(latest_block), latest_block);
+
+    info!("{} start_block is {}", event_name, start_block);
+
+    if let Some(end_block) = manifest_end_block {
+        if end_block > latest_block {
+            error!("{} - end_block supplied in yaml - {} is higher then latest - {} - end_block now will be {}", event_name, end_block, latest_block, latest_block);
+        }
+    }
+
+    let (end_block, indexing_distance_from_head) =
+        calculate_safe_block_number(reorg_safe_distance, &provider, latest_block, end_block)
+            .await?;
+
+    Ok((start_block, end_block, indexing_distance_from_head))
+}
+
 pub async fn start_indexing(
     manifest: &Manifest,
     project_path: &Path,
     dependencies: &[ContractEventDependencies],
     no_live_indexing_forced: bool,
     registry: Arc<EventCallbackRegistry>,
+    trace_registry: Arc<TraceCallbackRegistry>,
 ) -> Result<Vec<ProcessedNetworkContract>, StartIndexingError> {
     let start = Instant::now();
-
     let database = initialize_database(manifest).await?;
     let event_progress_state = IndexingEventsProgressState::monitor(&registry.events).await;
 
     // we can bring this into the yaml file later if required
     let semaphore = Arc::new(Semaphore::new(100));
+
     // need this to keep track of dependency_events cross contracts and events
     let mut event_processing_configs: Vec<Arc<EventProcessingConfig>> = vec![];
+
     // any events which are non-blocking and can be fired in parallel
     let mut non_blocking_process_events = Vec::new();
     let mut dependency_event_processing_configs: Vec<ContractEventsDependenciesConfig> = Vec::new();
+
     // if you are doing advanced dependency events where other contracts depend on the processing of
     // this contract you will need to apply the dependency after the processing of the other
     // contract to avoid ordering issues
     let mut apply_cross_contract_dependency_events_config_after_processing = Vec::new();
-
     let mut processed_network_contracts: Vec<ProcessedNetworkContract> = Vec::new();
 
+    for event in trace_registry.events.iter() {
```

Comment:
hmm not ideal this is blocking the registry.events we should probs make them run at same time so its not blocking anything

Reply:
I've extracted these out to two "sub-indexing" functions and run them concurrently 👍 

The only blocking operation in each was a single `await` to the `get_start_end_block` function, which would be in the order of a few hundred ms at most but even so I've run them concurrently now so should alleviate any additional latency this was likely to cause.

And when this is disabled it wont have any latency increase ofc